dietstory-routing/game_asset_manager/views.py

```diff
@@ -94,33 +97,120 @@ def get(self, request, *args, **kwargs):
+		except GameVersions.DoesNotExist:
+			game_version = None
 
-        elif self.request.method == "POST":
-            return (permissions.IsAdminUser(),)
+		if game_version:
+			return JsonResponse(
+				{'game_version': GameVersionSerializer(game_version).data},
+				status=status.HTTP_200_OK)
+		else:
+			return JsonResponse(
+				{'message': 'No version of the game exists.'},
+				status=status.HTTP_404_NOT_FOUND)
 
-    def get(self, request, *args, **kwargs):
+	def post(self, request, *args, **kwargs):
+		"""
+        summary: Post Game Version
+        description: Submit game version with live by time.
+        tags:
+            - GameVersionView
+        responses:
+            200:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Game version has been submitted successfully.
+            400:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Invalid input parameters.
+            500:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Game version submission was not successful.
+        """
+		game_version_form = SubmitGameVersionForm(request.data)
 
-        try:
-            game_version = GameVersions.objects.get(is_live=True)
-        except GameVersions.DoesNotExist:
-            game_version = None
+		if game_version_form.is_valid():
 
-        if game_version:
-            return JsonResponse(
-                {'game_version': GameVersionSerializer(game_version).data},
-                status=status.HTTP_200_OK)
-        else:
-            return JsonResponse(
-                {'message': 'No version of the game exists.'},
-                status=status.HTTP_404_NOT_FOUND)
+			try:
+				major_ver = game_version_form.cleaned_data.get('major_ver')
+				minor_ver = game_version_form.cleaned_data.get('minor_ver')
+				live_by = game_version_form.cleaned_data.get('live_by')
+
+				game_version = GameVersions(major_ver=major_ver, minor_ver=minor_ver, live_by=live_by)
+				game_version.save()
+
+				#add this job to the job scheduler
+				version_update_scheduler.add_job(update_game_version, 'date', run_date=game_version.live_by, args=[game_version.major_ver, game_version.minor_ver])
```

Comment:
we should also be allowed to submit versions with no live_by values. Ensure to not create a job for this case.

Reply:
Ok this is new, will add.